core/src/test/java/org/apache/calcite/test/JdbcAdapterTest.java

```diff
@@ -1147,6 +1153,52 @@ private LockWrapper exclusiveCleanDb(Connection c) throws SQLException {
     });
   }
 
+  @Test void testBatchNestedLoopJoinPlan() {
+    final String sql = "SELECT *\n"
+        + "FROM \"s\".\"emps\" A\n"
+        + "LEFT OUTER JOIN \"foodmart\".\"store\" B ON A.\"empid\" = B.\"store_id\"";
+    final String explain = "JdbcFilter(condition=[OR(=($cor0.empid0, $0), =($cor1.empid0, $0)";
+    final String jdbcSql = "SELECT *\n"
+        + "FROM \"foodmart\".\"store\"\n"
```

Comment:
Are there any other edge cases or situations we should test? Asked another way: Is one test enough to ensure that this works the way we intend?

Reply:
I think one test should be enough, as only one additional execution path is added to the code.

```java
    Context context = correlTableMap.get(variable.id);
    if (context != null) {
      return context;
    }
```